jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactory.java

```diff
@@ -35,13 +35,15 @@
     /**
      * <p>Make a Resource containing a collection of other resources</p>
      * @param resources multiple resources to combine as a single resource. Typically, they are directories.
-     * @return A Resource of multiple resources.
+     * @return A Resource of multiple resources or a single resource if only 1 is passed, or null if none are passed
      * @see ResourceCollection
      */
-    static ResourceCollection combine(List<Resource> resources)
+    static Resource combine(List<Resource> resources)
```

Comment:
`combine` should return a `Resource` and not a `ResourceCollection`, since if the passed list has only a single entry, then a collection is not needed.    The caller should not need to know if it is a collection or not.